<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Package reviews</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">FISH 549</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="syllabus.html">Syllabus</a>
</li>
<li>
  <a href="lectures.html">Schedule &amp; materials</a>
</li>
<li>
  <a href="homework.html">Assignments</a>
</li>
<li>
  <a href="references.html">References</a>
</li>
<li>
  <a href="https://github.com/FISH549">
    <span class="fab fa-github"></span>
     
    
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Package reviews</h1>

</div>


<p><br></p>
<p>For assignment #7 students were asked to provide a summary overview
of any package they enjoyed or used frequently. Here is a list of some
of their responses, provided with permission.</p>
<hr />
<div id="heel" class="section level1">
<h1><code>{HEEL}</code></h1>
<p><br></p>
<p>Contributed by G Henry</p>
<p><br></p>
<div id="location" class="section level2">
<h2>Location</h2>
<p><a href="https://github.com/gholtgrieve/HEEL"
class="uri">https://github.com/gholtgrieve/HEEL</a></p>
</div>
<div id="vignettes" class="section level2">
<h2>Vignette(s)</h2>
<p>This package does not have any vignettes.</p>
</div>
<div id="applications" class="section level2">
<h2>Application(s)</h2>
<p>There are no websites or applications from this package.</p>
</div>
<div id="review" class="section level2">
<h2>Review</h2>
<p>This package is a collection of functions that have been written by
members of the Holtgrieve Ecosystem Ecology Lab that generally are all
related to stable isotope data analysis. Personally, I most frequently
use the “EA.NACHO” functionality of the package, which takes raw stable
isotope analysis data and finalizes that data by comparing it with a
known standard value. It also makes graphs to check the accuracy and
precision of the mass spec and elemental analyzer and drift corrects the
data. It outputs a PDF file that has lots of helpful information about
the samples being analyzed. This package is easy to use and has lots of
good documentation to help the user understand how to use it and what it
is doing which I like. There is nothing specific that I wish it did
differently, but it also took a few versions to get it working how it
currently is. When I first started using this package it was still in
development and so some of the functionality didn’t work well yet. I
would only reccommend this package to people also using the SAFS EA and
mass spec because a lot of the functionality is specific to our
system.</p>
<hr />
</div>
</div>
<div id="dadjokeapi" class="section level1">
<h1><code>{dadjokeapi}</code></h1>
<p><br></p>
<p>Contributed by J Peraza</p>
<p><br></p>
<div id="location-1" class="section level2">
<h2>Location</h2>
<p>You can access this package from GitHub by installing:</p>
<pre class="r"><code># install.packages(&quot;devtools&quot;)
devtools::install_github(&quot;jhollist/dadjokeapi&quot;)</code></pre>
<p>And it is also on CRAN:</p>
<pre class="r"><code>install.packages(&quot;dadjokeapi&quot;)</code></pre>
</div>
<div id="vignettes-1" class="section level2">
<h2>Vignette(s)</h2>
<p>This package unfortunately has no vignettes, which is extremely
disappointing–I think this needs to change. But, there is a standard
help file when you run.</p>
<pre class="r"><code>?groan</code></pre>
</div>
<div id="applications-1" class="section level2">
<h2>Application(s)</h2>
<p>There is one application of this package, and it’s this incredible
website: <a href="https://icanhazdadjoke.com/"
class="uri">https://icanhazdadjoke.com/</a>. The package is used in this
website to generate a new dad joke every time you click the green button
“new joke” at the top of the webpage.</p>
<p>A basic application of the package:</p>
<pre class="r"><code>dadjokeapi::groan()</code></pre>
<p>Here is the generated joke:</p>
<p><strong>Today, my son asked “Can I have a book mark?” and I burst
into tears. 11 years old and he still doesn’t know my name is
Brian.</strong></p>
</div>
<div id="review-1" class="section level2">
<h2>Review</h2>
<p>The primary purpose of this package, as defined by the creator
themselves, is to “make you laugh in spite of yourself.” Whenever you
are feeling down, just generate a new dad joke for giggles. What I
really like about this package is its simplicity in that it is a single
function. A plus is the function being named “groan,” it can’t get
better than that. I do wish this package had more functions. For
example, maybe different functions to generate different dad jokes that
fall under certain categories like “one-liners,” “corny,”
“cringeworthy,” and “pun-tastic.” I think there’s real potential here. I
100% recommend this package for anyone who needs a dad joke on the
fly!</p>
<hr />
</div>
</div>
<div id="gganimate" class="section level1">
<h1><code>{gganimate}</code></h1>
<p><br></p>
<p>Contributed by T Code</p>
<p><br></p>
<div id="availability" class="section level2">
<h2>Availability</h2>
<p>Available on both CRAN (use
<code>install.packages("gganimate")</code> and on GitHub via
<code>install_github('thomasp85/gganimate')</code>.</p>
</div>
<div id="vignettes-2" class="section level2">
<h2>Vignettes</h2>
<p><code>{gganimate}</code> has a vignette that includes function
definitions, examples of code and the graph outcomes. It also explains
how to find your animation, and provides options for additional
rendering formats.</p>
</div>
<div id="applications-2" class="section level2">
<h2>Applications</h2>
<p>There are many blogs that describe the usefulness of gganimate
including posts from solarchemist, Medium, R-bloggers, and others that
highlight the usefulness of the package. I have seen used to show the
progression of COVID cases, the relationship between GDP and life
expectancy, and NYC housing trends overtime.</p>
</div>
<div id="review-2" class="section level2">
<h2>Review</h2>
<p>The purpose of this package is to add animation to an otherwise
static graphic. This package allows you animate points to easily show
how data changes over time. I like how it builds off of ggplot and is
easy to implement on graphs that are already coded. I like how much
online chatter there is about this package, it makes it easy to get
ideas on how to use it effectively. That being said, it seems that folks
online struggle with getting animations to work without additional
packages installed. I haven’t run into these errors but it shows that in
some instances, gganimate isn’t as easy as it seems. Overall, I would
definitely recommend this package to someone! The graphics produced by
this package would be perfect to include in a presentation or on a
project website.</p>
<hr />
</div>
</div>
<div id="camtrapr" class="section level1">
<h1><code>{camtrapR}</code></h1>
<p><br></p>
<p>Contributed by Anonymous</p>
<p><br></p>
<div id="location-2" class="section level2">
<h2>Location</h2>
<p>You can install the release version of <code>{camtrapR}</code> from
CRAN:</p>
<pre class="r"><code>install.packages(&quot;camtrapR&quot;)</code></pre>
<p>or the latest GitHub version (containing recent changes and new
features) via:</p>
<pre class="r"><code>remotes::install_github(&quot;jniedballa/camtrapR&quot;, build_vignettes = TRUE)</code></pre>
</div>
<div id="vignettes-3" class="section level2">
<h2>Vignette(s)</h2>
<p>The package has 5 formal vignettes that help users navigate it.</p>
<ol style="list-style-type: decimal">
<li><p>Organising raw camera trap images in camtrapR</p></li>
<li><p>Species and Individual Identification</p></li>
<li><p>Data Extraction from Images and videos, creating occupancy &amp;
secr input</p></li>
<li><p>Data exploration and visualisation</p></li>
<li><p>Multi-species occupancy models</p></li>
</ol>
</div>
<div id="applications-3" class="section level2">
<h2>Application(s)</h2>
<p>The camtrapR package has been used in a wide range of conservation
and research projects. For example, the Wildlife Conservation Society
has used it for monitoring and research projects relating to wildlife as
they use it to process and analyze camera trap data. The package has
also been used by researchers at the Smithsonian Institution. They
studied the distribution and abundance of large mammals in tropical
forests. There are other applications of this package, all demonstrating
that it can help analyze camera trap data and estimate parameters such
as density, abundance, and behavior of wildlife species.</p>
</div>
<div id="review-3" class="section level2">
<h2>Review</h2>
<p>The <code>{camtrapR}</code> package is a very useful tool for
analyzing camera trap data. This package contains a set of functions
that can be used to process, analyze, and visualize camera trap data. It
is designed to handle challenges associated with camera trap data
including missing data and repeated detections. I like that this package
is a comprehensive set of functions that are important for the type of
data I’m working with as it is very messy in nature. It contains
functions that will help me do my analysis including estimating animal
abundance, occupancy, activity patterns, and it can also help plot and
visualize that data. The documentation and vignettes associated with
this package are also very thorough and provide detailed explanations of
the functions, their parameters, and examples. As a beginner to this
type of camera data analysis, I find this package to be very helpful.
Because I haven’t spent much time working with this package, I still
don’t know if there any changes I’d like to see associated with this
package, but I think it has been relatively easy to learn compared to
other packages. Overall, I would recommend this package to anyone
working with camera trap data, as it provides a comprehensive set of
tools for processing and analyzing data.</p>
<hr />
</div>
</div>
<div id="ggplot2" class="section level1">
<h1><code>{ggplot2}</code></h1>
<p><br></p>
<p>Contributed by N Chambers &amp; Anonymous</p>
<p><br></p>
<div id="location-3" class="section level2">
<h2>Location</h2>
<p>You can install this package on both <a
href="https://cran.r-project.org/web/packages/ggplot2/index.html">CRAN</a>
and <a href="https://github.com/tidyverse/ggplot2">Github</a></p>
</div>
<div id="vignettes-4" class="section level2">
<h2>Vignette(s)</h2>
<p><code>{ggplot2}</code> has several vignettes in addition to standard
help files. It has three topics covered:</p>
<ol style="list-style-type: decimal">
<li><p>aesthetic specifications,</p></li>
<li><p>extending <code>{ggplot2}</code>, and</p></li>
<li><p>using <code>{ggplot2}</code> in packages.</p></li>
</ol>
<p>The first topic is probably the most commonly referred to vignette as
it covers the primary calls/arguments when using <code>{ggplot2}</code>
to plot.</p>
</div>
<div id="applications-4" class="section level2">
<h2>Application(s)</h2>
<p>Since this package is part of the Tidyverse it has a dedicated
section on the Tidyverse website. This website provides a lot of detail
with regard to how the functions contained in the package can be used.
It also has a section with 120 extensions to the package with examples
of what they do. One reason I feel the need to learn this package is
there are more examples on the internet of how to build figures with
ggplot than with base R. Thus there are thousands of websites and pages
giving examples of how to use this package to produce useful plots.</p>
<p>There is a website like the one we built, which can be found <a
href="https://ggplot2.tidyverse.org/">here</a>. This website includes a
variety of resources including a cheat sheet which can be pulled from
github <a
href="https://github.com/rstudio/cheatsheets/blob/main/data-visualization.pdf">here</a>.</p>
</div>
<div id="review-4" class="section level2">
<h2>Review</h2>
<p>This package is primarily used for plotting data and developing
figures for publications. I like that the package is no more difficult
to use or learn that base R plotting, and it uses less code overall to
produce similar figures. I also appreciate the amazing amount of support
this package has on online resources such as stack overflow and various
tutorial websites. It does a great job of being able to produce complex
and appealing figures while still maintaining its user friendly
functionality when producing exploratory plots. Since I have only used
it a few times I only have one complaint so far; I am not a big fan of
the grey background that is produced in plots by default. This is a
minor complaint however as it can be turned into a transparent
background easily. I have found the package to be easy to learn and I am
committed to learning to use this in favor of base R. At this point I
would reccomend this package to others, it seems to be the standard and
I have eoncountered no major downsides with several large upsides.</p>
<p>The primary purpose of the <code>{ggplot2}</code> package is for
graphics and visualizing data. This package improves on plotting in base
R in that you can easily call aesthetic arguments (e.g., black and
white) to create clean, easy-to-read figures. One of the most common
uses I have with <code>{ggplot2}</code> is its faceting capabilities.
With one easy line, <code>facet_wrap()</code>, I can break up data into
groups and visualize each group side-by-side in subplots. Generally,
this package was pretty easy to learn in its basic form. However,
learning to customize my figures using <code>{ggplot2}</code> took more
patience and skill. For example, custom colors can be tricky depending
on what you want. I have also found that a <code>{ggplot2}</code> call
for a single plot can very quickly become dozens of lines of code which
becomes a bit cumbersome to edit/review, especiall if annotated with
comments. In any case, I think there is a lot of utility in this
package, and I have and will continue to recommend this package to other
R users.</p>
<hr />
</div>
</div>
<div id="vcfr" class="section level1">
<h1><code>{vcfR}</code></h1>
<p><br></p>
<p>Contributed by Anonymous</p>
<p><br></p>
<div id="location-4" class="section level2">
<h2>Location</h2>
<p>This package is located on Github at <a
href="https://github.com/knausb/vcfR"
class="uri">https://github.com/knausb/vcfR</a> as well as CRAN.</p>
</div>
<div id="vignettes-5" class="section level2">
<h2>Vignette(s)</h2>
<p>This package has not one but four formal vignettes. These include a
vignette on what a VCF (Variant Call Format) file is, one on a general
intro to <code>{vcfR}</code>, one on the workflow for
<code>{vcfR}</code>, and one for converting objects in
<code>{vcfR}</code> to other data or file formats.</p>
</div>
<div id="applications-5" class="section level2">
<h2>Application(s)</h2>
<p>There are two thorough applications of this package that are both
linked in the README. One is an entire website devoted to the package,
covering everything from background information on R and file types used
(they give a nice background as to what a VCF file is and how it’s
structured) as well as many tutorials as to analyses <code>{vcfR}</code>
can be used for. In addition, they link another website they’ve build
all about population genetics and genomics in R, giving a really nice
work through of how you can use R to perform all sorts of filtering,
visualization, and analyses for population genetic/genomic studies. I
was really impressed by all this documentation.</p>
</div>
<div id="review-5" class="section level2">
<h2>Review</h2>
<p><code>{vcfR}</code> is described as a package to “help visualize,
manipulate and quality filter VCF data”. Genetic/genomic data is often
stored in VCF files and there exist many tools for filtering,
manipulating, and analyzing these data. However, a disadvantage of many
of these tools, such as VCFtools, is that they are run in Linux, making
any sort of code documentation or data visualization difficult. This can
make setting thresholds for filtering difficult as well.
<code>{vcfR}</code> allows us to visualize this data though and keep
records of code and previous filtering steps, allowing for better
threshold determination and more consistent filtering across different
datasets. An aspect I especially appreciate is how easy it is to output
these data to other file formats that may be necessary for additional
analyses. The authors of this package did a fantastic job providing all
sorts of tutorials and documentation on the applications this package
has, as well as how R can be utilized in population genetic/genomic
studies in general. This makes learning how to use this package and its
various utilities very easy to learn.</p>
<p>The only aspect of this package I’m not a huge fan of is that the
functions in <code>{vcfR}</code> are designed to only work on a single
chromosome. When you have data from across the genome, it can be tedious
to have to break data down chromosome by chromosome, especially in
organisms like salmonids, which can have chromosome numbers in the high
twenties. It would be nice if <code>{vcfR}</code> could be expanded to
work with data across the genome simultaneously. Overall though I would
recommend <code>{vcfR}</code> to anyone working with genetic/genomic
data.</p>
<hr />
</div>
</div>
<div id="wesanderson" class="section level1">
<h1><code>{wesanderson}</code></h1>
<p><br></p>
<p>Contributed by N Doran</p>
<p><br></p>
<div id="locations" class="section level2">
<h2>Location(s)</h2>
<p>The <code>{wesanderson}</code> package has color palettes based on
Wes Anderson films. It is on both GitHub and CRAN.</p>
<pre class="r"><code>install.packages(&quot;wesanderson&quot;)</code></pre>
</div>
<div id="vignettes-6" class="section level2">
<h2>Vignette(s)</h2>
<p>The package does have a vignette but it is not very complete. The
most helpful way to learn about applications of this package are on the
Github repo: <a href="https://github.com/karthik/"
class="uri">https://github.com/karthik/</a><code>{wesanderson}</code></p>
</div>
<div id="applications-6" class="section level2">
<h2>Application(s)</h2>
<p>In the description it has the functions needed to see the palette
names and see the different palettes. One of the applications it
mentions is the Zissou1 palette (inspired by the Life Aquatic with Steve
Zissou) is good for creating heat maps. The end of the description also
has a list of examples of this palette being used in publications. There
are more applications of it on R bloggers: <a
href="https://www.r-bloggers.com/2022/07/colorful-r-plots-with-wes-anderson-palettes-pirate-ships/"
class="uri">https://www.r-bloggers.com/2022/07/colorful-r-plots-with-wes-anderson-palettes-pirate-ships/</a></p>
<p>Most palettes have only 4 to 5 colors in them, but you can include an
argument in the function wespalette() to create a continuous gradient to
pull more colors from, which is demonstrated in examples in the
description file.</p>
<pre class="r"><code>names(wes_palettes)</code></pre>
<p>My favorite palette is probably the Royal1 palette inspired by the
<em>Royal Tenenbaums</em>. I’ve used this palette as inspiration for
posters and powerpoint presentations because I like the gradient of warm
colors with the blue accent color. Taking the red from Ben Stiller’s
tracksuit is a nice touch.</p>
<pre class="r"><code>wes_palette(&quot;Royal1&quot;)</code></pre>
</div>
<div id="review-6" class="section level2">
<h2>Review</h2>
<p>I’m not familiar with many different R packages so I thought this
would be a fun one to highlight after the lecture we had on graphic
design and color palettes. As a film nerd I always really liked how
color is used in Wes Anderson films and these palettes capture the
aesthetics of some of my favorite movies, so that’s fun. It doesn’t look
like the newest film French Dispatch has been added, so I would like to
see that because one of the skits has a nice color palette of pink,
blue, and yellow that aren’t as pastel-like as the Moonrise Kingdom
palette with similar colors.</p>
<pre class="r"><code>wes_palette(&quot;Moonrise3&quot;)</code></pre>
<p>I don’t really have anything else to say–they’re color palettes. They
are easy to use in both base R and <code>{ggplot}</code>, many are
color-blind friendly. It’s just fun to have cultural references in data
science.</p>
<hr />
</div>
</div>
<div id="patchwork" class="section level1">
<h1><code>{patchwork}</code></h1>
<p><br></p>
<p>Contributed by K Veggerby</p>
<p><br></p>
<div id="location-5" class="section level2">
<h2>Location</h2>
<p>The package is on CRAN and can be installed in RStudio using the
following R command:</p>
<pre class="r"><code>install.packages(&quot;patchwork&quot;)</code></pre>
</div>
<div id="vignettes-7" class="section level2">
<h2>Vignette(s)</h2>
<p>There is a <a
href="https://cran.r-project.org/web/packages/patchwork/vignettes/patchwork.html">short
vignette</a> going through the basics of patchwork syntax. This is
helpful if you are new to patchwork and want to see how it works. The
idea is to combine plots, figures, maps etc in R as simply as possible.
It also contains many options for more complex or detailed plot
combinations. For more detailed examples see: <a
href="https://patchwork.data-imaginist.com/"
class="uri">https://patchwork.data-imaginist.com/</a></p>
<p>If you’re interested in learning patchwork, I would spend about 5
minutes on the cran page familiarizing yourself with the basics, and
then spend the rest of your time on the package website listed
above.</p>
</div>
<div id="applications-7" class="section level2">
<h2>Application(s)</h2>
<p><a href="https://patchwork.data-imaginist.com/"
class="uri">https://patchwork.data-imaginist.com/</a></p>
<p>This website built by the developers goes through many examples of
how to use patchwork mainly using the mtcars dataset in R.</p>
</div>
<div id="review-7" class="section level2">
<h2>Review</h2>
<p>The purpose of patchwork is to combine plots, maps, pictures etc. in
R into multiplots as simply as possible. The syntax is such that it
visually makes sense to the human eye.</p>
<p>For instance:</p>
<pre class="r"><code>plot1 + plot2 </code></pre>
<p>produces a figure that has plot1 and plot2 side by side.</p>
<pre class="r"><code>plot1 / plot2 </code></pre>
<p>produces a figure that has plot1 and plot2 stacked vertically.</p>
<p>Beyond the simplicity of the commands and syntax, the package has a
huge list of options to customize multiplots and make small tweaks to
customize pretty much every aspect of a figure. Different commands can
be added on by adding a <code>+</code> and another line, in the same way
that ggplot adds additional items. Patchwork accepts most R objects as
inputs. For instance, you could generate a series of plots in ggplot,
and then add them all together with patchwork.</p>
<p>Example: <code>ggplot1</code>, <code>gis.shape.file</code>,
<code>ggplot3</code>, and <code>png.object</code> could be made into a 2
x 2 multiplot using simply:</p>
<pre class="r"><code>Cool_plot &lt;- (ggplot1 + gis.shape.file) / (ggplot3 + png.object)</code></pre>
<p>There are a number of other plot combine packages out there like
<code>{grid}</code> and <code>{cowplot}</code>. While these can be
helpful, patchwork generally can do everything that other packages can
do, but it does them better or more efficiently with less code. In fact,
the creator of (I believe) cowplot has stated that people should just
use patchwork for 90% of their plot combination needs.</p>
<p>Given how easy patchwork is to use and understand, the customization,
and the large amount of tutorial/documentation, this should be
everyone’s go-to plot combination package. Along with other packages
such as ggplot and Rcolorbrewer, patchwork forms part of a core group of
data visualization packages that everyone should know.</p>
<hr />
</div>
</div>
<div id="ss3sim" class="section level1">
<h1><code>{ss3sim}</code></h1>
<p><br></p>
<p>Contributed by Anonymous</p>
<p><br></p>
<div id="location-6" class="section level2">
<h2>Location</h2>
<p>The package can be downloaded from Cran or from Github.</p>
<ol style="list-style-type: decimal">
<li>Cran</li>
</ol>
<pre class="r"><code>install.packages(&#39;ss3sim&#39;,
                 repos=&#39;https://cran.r-project.org/web/packages/ss3sim/index.html&#39;,
                 dependencies=TRUE)</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Github</li>
</ol>
<pre class="r"><code>remotes::install_github(&quot;ss3sim/ss3sim&quot;,
                        ref = &quot;main&quot;,
                        build_vignettes = TRUE, dependencies = TRUE )</code></pre>
</div>
<div id="vignettes-8" class="section level2">
<h2>Vignette(s)</h2>
<p><code>{ss3sim}</code> has a number of useful help files.</p>
<p>There are three documents which are useful when working with
<code>{ss3sim}</code>. The first is the <code>README.md</code>, which is
comprehensive and offers notes on the installation, file structure and
an example simulation of the package. The <code>README.md</code> can be
found <a
href="https://github.com/ss3sim/ss3sim/blob/main/README.md">here</a>.</p>
<p>There are three vignettes available on the github which cover</p>
<ol style="list-style-type: decimal">
<li><p><a
href="https://ss3sim.github.io/ss3sim/articles/introduction.html">Introduction</a></p></li>
<li><p><a
href="https://github.com/ss3sim/ss3sim/blob/main/vignettes/making-models.Rmd">Making
models</a></p></li>
<li><p><a
href="https://github.com/ss3sim/ss3sim/blob/main/vignettes/modifying-models.Rmd">Modifying
models</a>.</p></li>
</ol>
<p>The package is actively under development by NOAA, and as such the <a
href="https://github.com/ss3sim/ss3sim/discussions">discussion
board</a>,on GitHub is particularly useful for resolving queries.</p>
<p>There is also an introduction <a
href="https://cran.r-project.org/web/packages/ss3sim/vignettes/introduction.html">avilable
on CRAN</a>, but it is deprecated and following the instructions will
not allow the user to create simulations. This is particularly
problematic as when Google is used to <a
href="https://www.google.com/search?q=introduction+to+ss3sim&amp;oq=introduction+to+ss3sim&amp;aqs=chrome..69i57j69i60l2.4403j0j7&amp;sourceid=chrome&amp;ie=UTF-8">search
for introductory material</a>, the CRAN intro is the first result.</p>
</div>
<div id="applications-8" class="section level2">
<h2>Application(s)</h2>
<p>The primary application of the package is that it allows users to
crate simulated fish populations and estimate them using the <a
href="https://nmfs-stock-synthesis.github.io/doc/SS330_User_Manual.html">Stock
Synthesis 3</a> modelling framework.The package has been employed for
simulation testing documented in peer-review literature. Notable
publications include:</p>
<p>Monnahan et al. (<a href="#ref-monnahan_effect_2016">2016</a>)<br />
Ono et al. (<a href="#ref-ono_importance_2015">2015</a>)<br />
Rudd et al. (<a href="#ref-rudd_catch_2021">2021</a>)</p>
</div>
<div id="review-8" class="section level2">
<h2>Review</h2>
<p><code>{ss3sim}</code> is an R package which allows users to simulate
fish stock populations and asses them using <a
href="https://nmfs-stock-synthesis.github.io/doc/SS330_User_Manual.html">Stock
Synthesis 3</a> modelling framework. The package was initially developed
as part of a graduate class project in the school of Aquatic and
fisheries Science, University of Washington, with the project being led
by Sean Anderson (Anderson et al. (<a
href="#ref-anderson_ss3sim_2014">2014</a>)). Each simulation has three
components an Operating model (OM) which represent the true population
such as natural mortality <em>M</em>, recruitment <em>R</em> and fishing
mortality <em>F</em>,the Estimation method (EM) (Stock assessments,
survey data etc) which denotes how the population is to be assessed and
the control file, which allows the users how and when they would like to
alter the EM and OM.</p>
<p>While the file structure of the does appear cumbersome at first,
running a pre-specification (or boiler plate) is realities
straightforward by following the introduction vignette. The package
allows users to alter SS3 assessments in a relatively straightforward
fashion,compared to editing the large .text files which SS3 assessments
use. The other nice aspect of the package is there are three boiler
plates, each mimicking a different fish life-history pattern. The
species are Cod <em>Gadus morhua</em>, pacific anchovy <em>Engraulis
mordax</em> and yellowtail flounder <em>Pleuronectes
ferruginea</em>.</p>
<p>I would recommend the package to anyone interested in fisheries
simulation testing or with <code>{ss3sim}</code>. As a user who had no
SS3 experience, I was able to alter the boiler plates and simulate the
effect of having <em>F</em> set far higher than is suitable. The effect
of this on a fishery over a time period (I chose 100 years) could then
be observed.</p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<p>Anderson, S.C., Monnahan, C.C., Johnson, K.F., Ono, K., Valero, J.L.,
2014. ss3sim: An R Package for Fisheries Stock Assessment Simulation
with Stock Synthesis. PLoS ONE 9, e92725. <a
href="https://doi.org/10.1371/journal.pone.0092725"
class="uri">https://doi.org/10.1371/journal.pone.0092725</a></p>
<p>Monnahan, C.C., Ono, K., Anderson, S.C., Rudd, M.B., Hicks, A.C.,
Hurtado-Ferro, F., Johnson, K.F., Kuriyama, P.T., Licandeo, R.R.,
Stawitz, C.C., Taylor, I.G., Valero, J.L., 2016. The effect of length
bin width on growth estimation in integrated age-structured stock
assessments. Fisheries Research, Growth: Theory, estimation, and
application in fishery stock assessment models 180, 103–112. <a
href="https://doi.org/10.1016/j.fishres.2015.11.002"
class="uri">https://doi.org/10.1016/j.fishres.2015.11.002</a></p>
<p>Ono, K., Licandeo, R., Muradian, M.L., Cunningham, C.J., Anderson,
S.C., Hurtado-Ferro, F., Johnson, K.F., McGilliard, C.R., Monnahan,
C.C., Szuwalski, C.S., Valero, J.L., Vert-Pre, K.A., Whitten, A.R.,
Punt, A.E., 2015. The importance of length and age composition data in
statistical age-structured models for marine species. ICES Journal of
Marine Science 72, 31–43. <a
href="https://doi.org/10.1093/icesjms/fsu007"
class="uri">https://doi.org/10.1093/icesjms/fsu007</a></p>
<p>Rudd, M.B., Cope, J.M., Wetzel, C.R., Hastie, J., 2021. Catch and
Length Models in the Stock Synthesis Framework: Expanded Application to
Data-Moderate Stocks. Frontiers in Marine Science 8, 663554. <a
href="https://doi.org/10.3389/fmars.2021.663554"
class="uri">https://doi.org/10.3389/fmars.2021.663554</a></p>
<hr />
</div>
</div>
<div id="fd" class="section level1">
<h1><code>{FD}</code></h1>
<p><br></p>
<p>Contributed by E Bishop</p>
<p><br></p>
<div id="location-7" class="section level2">
<h2>Location</h2>
<p>You can find this package on <a
href="https://cran.r-project.org/web/packages/FD/index.html">CRAN</a>
and also on <a href="https://github.com/cran/FD">Github</a> as a
read-only mirror of the CRAN repository.</p>
</div>
<div id="vignettes-9" class="section level2">
<h2>Vignette(s)</h2>
<p>There are no vignettes as part of this package.</p>
</div>
<div id="applications-9" class="section level2">
<h2>Application(s)</h2>
<p>I haven’t found any applications of this package yet outside of
scientific publications. It’s a relatively new package.</p>
</div>
<div id="review-9" class="section level2">
<h2>Review</h2>
<p>The primary purpose of this package is for community ecologists to
analyze functional trait data. Functional traits are characteristics of
species that describe the ecological niche they occupy within a
community. Once a user has created matrices of site x species and
species x trait data they can use this package to calculate several
diversity indices using the <span class="math inline">\(dbFD\)</span>
function. For example, these indices include functional richness,
evenness, and divergence, which are all aspects of alpha (within
community) diversity. The diversity indices are based on a Principal
Coordinate Analysis (PCoA) and users can extract the PCoA coordinates
from the <span class="math inline">\(dbFD\)</span> output to visualize
the ordination. I like this package because the output of one function
gives quite a bit of information about a community dataset. I do,
however, wish there was more documentation and more help files to make
it more user-friendly. You have to have your input matrices formatted in
a very particular way, and there’s no examples online of how to do this,
so using the package involves a lot of trial and error therefore making
it a little tricky to learn how to use. I would recommend this package
because I think the output is super interesting, but in recommending the
package to a friend I would also share my R code with them so they could
have an easier time getting started.</p>
<hr />
</div>
</div>
<div id="siber" class="section level1">
<h1><code>{SIBER}</code></h1>
<p><br></p>
<p>Contributed by C Backstrom</p>
<p><br></p>
<div id="location-8" class="section level2">
<h2>Location</h2>
<p>You can download the Stable Isotope Bayesian Ellipses in R
(<code>{SIBER}</code>) package, created by Dr. Andrew Jackson and
Dr. Andrew Parnell, on the web both via CRAN (<a
href="https://cran.r-project.org/web/packages/SIBER/index.html"
class="uri">https://cran.r-project.org/web/packages/SIBER/index.html</a>)
and via Andrew Jackson’s Github repository (<a
href="https://github.com/AndrewLJackson/SIBER"
class="uri">https://github.com/AndrewLJackson/SIBER</a>).</p>
</div>
<div id="vignettes-10" class="section level2">
<h2>Vignette(s)</h2>
<p>The CRAN repository for the <code>{SIBER}</code> package offers 11
vignettes demonstrating different features of the R code calculations
and their interpretation. Topics covered in these vignettes range from
vector statistics on stable isotope biplot centroids to plot
customization to differences between comparing community and population
isotope biplots.</p>
</div>
<div id="applications-10" class="section level2">
<h2>Application(s)</h2>
<p>The <code>{SIBER}</code> package can be used to calculate the overlap
in isotopic niche space (e.g., biplots of carbon-13 and nitrogen-15
signatures) between two or more groups of organisms (such as species,
populations, or communities) to determine the trophic
similarity/difference between these groups. <code>{SIBER}</code> biplots
have appeared in scientific articles in various areas of biology and
ecology since the initial creation of the package in 2011.</p>
</div>
<div id="review-10" class="section level2">
<h2>Review</h2>
<p>The primary purpose of the <code>{SIBER}</code> package is to compare
the trophic state of different groups of organisms (e.g., different
populations or species) by generating biplots of carbon and nitrogen
stable isotopic signatures and producing Bayesian ellipses to
statistically compare the overlap in signatures between those different
groups. As a coral ecophysiologist, I use <code>{SIBER}</code> to
quantify the overlap in isotopic signatures between corals and their
algal symbionts, interpreting higher percent overlap as a higher
exchange of resources (i.e., a higher degree of mutualism) between these
two members of coral symbiosis (see <a
href="https://www.science.org/doi/10.1126/sciadv.aaz5443">Conti-Jerpe et
al. (2020)</a>.</p>
<p>I really appreciate how easy it is to use this package for both
plotting and statistical analysis of isotopic data. The vignettes
provide extensive comments explaining every code line and its
interpretation, and the generated isotopic biplots are easy to
manipulate using your own code. The <code>{SIBER}</code> model runs
relatively quickly (in my experience, being able to process 4000
Bayesian simulations of sample sets of ~50 per each of 2 groups of data
in just a few minutes) and the associated Hotelling’s T2 and centroid
data for the ellipses are relatively easy to interpret. Having first
learned to use <code>{SIBER}</code> without much background in R or R
Studio, I can attest that the package is easy to learn to use, and that
the package creators are available by email to answer any further
questions.</p>
<p>My only warning for using the <code>{SIBER}</code> package would be
to keep up to date with the newest discussions of the limitations and
applications of the model itself. Especially in coral-centric fields,
<code>{SIBER}</code> applications are still in their infancy, and many
isotopic experts (including those reviewing my manuscripts) often have
nuanced interpretations of the results that may not be evident from the
previous literature.</p>
<hr />
</div>
</div>
<div id="tidyr" class="section level1">
<h1><code>{tidyr}</code></h1>
<p><br></p>
<p>Contributed by Anonymous</p>
<p><br></p>
<div id="location-9" class="section level2">
<h2>Location</h2>
<p>This package can be accessed by installing Tidyverse as a whole via
CRAN: <code>install.packages("tidyverse")</code>.</p>
<p>Alternatively, the singular package can be installed with
<code>install.packages("tidyr")</code>.</p>
<p>The development version of package can also be installed via GitHub
with <code>devtools::install_github("tidyverse/tidyr")</code>.</p>
</div>
<div id="vignettes-11" class="section level2">
<h2>Vignette(s)</h2>
<p>There are 5 categories of functions within the package and they are
listed on the <a href="https://tidyr.tidyverse.org/index.html">Tidyverse
website.</a>. Each category has extended guides to describe the
functions within.</p>
<ul>
<li>Pivoting</li>
<li>Rectangling</li>
<li>Nesting</li>
<li>Splitting and Combining</li>
<li>Making Implicit Values Explicit</li>
</ul>
</div>
<div id="applications-11" class="section level2">
<h2>Application(s)</h2>
<p>The goal of <code>{tidyr}</code> is in the title–it is meant to help
you tidy your data! <a href="https://tidyr.tidyverse.org/">The
website</a> provides a cheatsheet and further explanation of various
situations that it may be applicable in.</p>
</div>
<div id="review-11" class="section level2">
<h2>Review</h2>
<p>I learned about <code>{tidyr}</code> when taking a very simple
Introduction to R course in the final year of my undergraduate career. I
was briefly reintroduced to the package in FISH 552/553 and I was glad
to be reminded about a tool that could aid in the time-consuming process
of cleaning one’s data. Because 90% of research is data wrangling, I
find that this package can be incredibly useful despite having yet to
use it personally. Particularly, it would have been applicable for the
dense spatial data of my undergraduate research and made the process of
managing more complex data more straightforward. As an individual who is
still relatively new to R and coding languages, I find it enticing to
streamline a workflow.</p>
<p>The package highlights the two main properties of tidy data: each
column is a variable and each row is an observation. This concept is
ideal for consistency in your coding workflow because it arranges your
data in the same manner. The structure ultimately enables you to push
your data through various functions without worrying about
compatibility.</p>
<p><code>{tidyr}</code> is particular useful for:</p>
<ul>
<li>Changing the shape of a data set</li>
<li>Changing the hierarchy of data set</li>
<li>Handling deeply nested data sets</li>
<li>Working with missing values (explicit and implicit)</li>
</ul>
<hr />
</div>
</div>
<div id="purrr" class="section level1">
<h1><code>{purrr}</code></h1>
<p><br></p>
<p>Contributed by Anonymous</p>
<p><br></p>
<div id="location-10" class="section level2">
<h2>Location</h2>
<p>You can install <code>{purrr}</code> from CRAN with</p>
<pre class="r"><code>install.packages(&quot;purrr&quot;)</code></pre>
</div>
<div id="vignettes-12" class="section level2">
<h2>Vignette(s)</h2>
<p>The package does have a formal vignette. It covers topics such as
what tools it draws inspiration from as well as the goals of
<code>{purrr}</code>.</p>
</div>
<div id="applications-12" class="section level2">
<h2>Application(s)</h2>
<p><span class="citation">@joshmccrain</span> on Twitter posted an
introduction and application of the package showing its basic functions
and how to create models and visualizations with <code>{purrr}</code> In
his demonstration he created a list of <code>{ggplot}</code> objects to
apply to a map, which gives you the ability to call them separately.</p>
</div>
<div id="review-12" class="section level2">
<h2>Review</h2>
<p>I don’t have too much experience with <code>{purrr}</code>, but from
what I’ve used it for and read about it online I find it really
interesting. The package is meant to provide a set of functions that can
make your coding more efficient for working with functions and vectors
in R. For example the map function completes the same tasks as a for
loop but with different coding, like the different R apply functions.
Though <code>{purrr}</code> functions are named more consistently than
apply functions and can make code more readable. I like the consistency
of this package, lambda functions are needed as much so it makes
navigating data manipulation easier and more readable/reproducible for
me. While I’m still learning all of the possibilities of the package, I
do not find it hard to figure out, there is very good documentation
available for it and lots of online resources for different applications
of the function. I don’t necessarily have any complaints for the
package, as it applications for what I have used are simple. I would
recommend the <code>{purrr}</code> package to those who do not like the
application of the apply function family in R, as <code>{purrr}</code>
can be easier to read and use, though if you are already used to the
apply functions, <code>{purrr}</code> may not be worth learning.</p>
<hr />
</div>
</div>
<div id="tidyverse" class="section level1">
<h1><code>{tidyverse}</code></h1>
<p><br></p>
<p>Contributed by Anonymous</p>
<p><br></p>
<div id="install-from-cran" class="section level2">
<h2>Install from CRAN</h2>
<pre class="r"><code>install.packages(&quot;tidyverse&quot;)</code></pre>
</div>
<div id="vignettes-13" class="section level2">
<h2>Vignettes</h2>
<p><a
href="https://cran.r-project.org/web/packages/tidyverse/vignettes/manifesto.html">The
Tidyverse Manifesto</a></p>
<p><a
href="https://cran.r-project.org/web/packages/tidyverse/vignettes/paper.html">Welcome
to the Tidyverse</a></p>
</div>
<div id="application" class="section level2">
<h2>Application</h2>
<p>The primary website for the package can be found <a
href="https://tidyverse.tidyverse.org/">here</a>. There are plenty of
online examples for the applications of the package (which is
essentially a collection of packages). This book is one example of a
pretty comprehensive reference:</p>
<p><a href="https://r4ds.had.co.nz/introduction.html">R for Data Science
by Hadley Wickham and Garrett Grolemund</a></p>
<p>Additionally, there are plenty of “cheat sheets” for <a
href="https://statsandr.com/blog/files/ggplot2-cheatsheet.pdf">plotting</a>
and <a
href="https://www.rstudio.com/wp-content/uploads/2015/02/data-wrangling-cheatsheet.pdf">data
wrangling</a>, and much more that can be found with a google search!</p>
</div>
<div id="review-13" class="section level2">
<h2>Review</h2>
<p>The <code>{tidyverse}</code> package is actually a collection of a
variety of useful, harmonious packages and is meant to simplify the
installation and use those packages. According to their website, the
<code>{tidyverse}</code> is “an opinionated collection of R packages
designed for data science. All packages share an underlying design
philosophy, grammar, and data structures.” The <code>{tidyverse}</code>
package and associated methodologies essentially streamline data
manipulation and visualization in a way that I find more intuitive and
useful as an ecologist (who is also definitely not a coding guru).</p>
<hr />
</div>
</div>
<div id="forcats" class="section level1">
<h1><code>{forcats}</code></h1>
<p><br></p>
<p>Contributed by E Jameson</p>
<p><br></p>
<div id="location-11" class="section level2">
<h2>Location</h2>
<p>This is a popular tidyverse package and is installed when you install
the package <code>{tidyverse}</code>. The majority of the information
comes from the <code>{forcats}</code> website found here <a
href="https://forcats.tidyverse.org/"
class="uri">https://forcats.tidyverse.org/</a></p>
<p>The following R code will install the package in R</p>
<pre class="r"><code># The easiest way to get forcats is to install the whole tidyverse:
install.packages(&quot;tidyverse&quot;)

# Alternatively, install just forcats:
install.packages(&quot;forcats&quot;)

# Or the the development version from GitHub:
# install.packages(&quot;devtools&quot;)
devtools::install_github(&quot;tidyverse/forcats&quot;)</code></pre>
</div>
<div id="vignettes-14" class="section level2">
<h2>Vignettes</h2>
<p>Yes, there are great vignettes for the <code>{forcats}</code>
package. The following article provides examples for the following
functions within the <code>{forcats}</code> package.</p>
<pre class="r"><code>fct_infreq()
fct_na_value_to_level()
fct_na_level_to_value()
fct_lump()
fct_reorder()
fct_shuffle()
fct_relevel()</code></pre>
</div>
<div id="application-1" class="section level2">
<h2>Application</h2>
<p>The website linked above provides a lot of different resources
including a cheat sheet <a
href="https://raw.githubusercontent.com/rstudio/cheatsheets/main/factors.pdf"
class="uri">https://raw.githubusercontent.com/rstudio/cheatsheets/main/factors.pdf</a></p>
<p>The first example uses the starwars dataset to create a table
displaying the number of characters that fall into the top 3 species
using <code>fct_lump()</code>. The second example creates a bar chart
displaying the number of characters with different eye colors. They then
sort the bars for aesthetics by reordering the factors with
<code>fct_infreq()</code>.</p>
</div>
<div id="review-14" class="section level2">
<h2>Review</h2>
<p>This package was created to make working with factor variables
easier. I used this package in a previous research project which
involved several factor variables such as plant species, level of
invasiveness, and site. I most commonly used it to reorder factors or
rename the factor levels for figures and models. I like the easy-to-read
function names. It is easy to figure out what they do based on their
title. Additionally, as part of the tidyverse this package has very good
documentation and examples. The combination of easy-to-read function
names and the supporting materials make this package very easy to learn.
I have not used the package in a while, but I don’t remember
encountering any challenges that were not quickly remedied with the
existing documentation. I that you could use base R to accomplish
similar tasks and therefore the package may be redundant. Overall,
however, I would definitely recommend using this package to someone
else. Especially as the hex sticker is adorable.</p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
